sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -191,7 +253,56 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
       ManagedTable
     }
 
-    client.createTable(
+    val hiveSerDe = HiveSerDe.sourceToSerDe(provider, hive.hiveconf, false).getOrElse(null)
+
+    if (hiveSerDe == null) {
+      logWarning(s"Unable to find the SerDe info for $provider, " +
+        s"the table probably unable be read by Hive")
+    }
+
+    val hiveTable = if (conf.writeSchemaToHiveMetastore && hiveSerDe != null) {
+      // get the schema from the data source
+      val ds = ResolvedDataSource(hive, userSpecifiedSchema, partitionColumns, provider, options)
+      ds.relation match {
+        case fs: HadoopFsRelation if fs.paths.length == 1 =>
```

Comment:
This is probably a reason that we need the configuration key "spark.sql.hive.writeDataSourceSchema"